### DISCUSSION TITLE
Re-localize an existing key

### DISCUSSION THREAD

**Discussion by laud** [FlineDev/RemafoX#102] [Help / FAQ]
Hi,

First off, great translation software. I'm loving it already.

If I already created a key with some text and I want to alter the text for the development language and all project localizations, what is the best way to do this? 

I could remove the key in the development localization, build, reinsert the key/text pair, and then translate again. However, this seems a bit of an awkward workflow. I can also run a re-translation of all values, but I just want to re-translate one string. This process seems wasteful and slow for large projects.

Is there a more direct way to doing a re-translation of a single string that has already been previously translated?

**Comment by Jeehut** [comment]
Hi @laud, thank you for asking that question.

I currently have 2 answers to this, I hope they help:
1. I recently shipped an "override same key" feature (see [here](https://github.com/FlineDev/RemafoX/issues/81#issuecomment-1509423434)) which allows you to simply act like you never created the key in the first place: Write your new String in code, mark it & use the add translation shortcut like normal, and simply choose the exact same key as the one you want to override. In that window, you can re-translate and even verify translations (like always) and by pressing "Insert" RemafoX will override the existing translations with the new one.
2. There's a feature planned to provide a dedicated Strings file editor with a nice & convenient UI. It will also include a machine-translate button, which will make it easy to do exactly what you describe above without having to do anything within code. Make sure to upvote that feature so I can prioritize accordingly: https://github.com/FlineDev/RemafoX/issues/12

Thank you for the kind words, I'm glad you're liking RemafoX! Keep your questions and feature requests incoming. With the [monthly feature updates](https://remafox.app/changelog/), I will try to make RemafoX the best tool it can be.
